lib/std/c.zig

```diff
@@ -3962,8 +3962,8 @@ pub const sigval = switch (native_os) {
     else => void,
 };
 
-pub const addrinfo = switch (native_os) {
-    .linux, .emscripten => linux.addrinfo,
+pub const addrinfo = blk: switch (native_os) {
+    .linux, .emscripten => if (builtin.abi.isAndroid()) continue :blk .freebsd else linux.addrinfo,
```

Comment:
```suggestion
    .linux => if (builtin.abi.isAndroid()) continue :blk .freebsd else linux.addrinfo,
    .emscripten => linux.addrinfo,
```

`*-emscripten-android` is not actually a valid target, so I suggest splitting these up.

Reply:
Actually, on second thought, I'm not loving the `continue` usage here. :thinking: I think it'd be better to just do `if (builtin.abi.isAndroid()) ... else switch (native_os) ...`. It results in some duplicate code, but I don't actually think that's a bad thing considering it's a completely different libc API from what the BSDs use; any similarity is incidental.